modin/experimental/engines/omnisci_on_native/frame/data.py

```diff
@@ -723,12 +723,12 @@ def astype(self, col_dtypes, **kwargs):
             The new frame.
         """
         columns = col_dtypes.keys()
-        new_dtypes = self.dtypes.copy()
+        new_dtypes = self._dtypes.copy()
```

Comment:
why do we want to use `_dtype` here? pls explain

Reply:
`_dtypes` is an internal attribute holding data type for all columns including index ones. `dtypes` is a public property holding data type for `columns`, which doesn't include index columns. Until recent times `dtypes` property worked incorrectly and always returned `_dtypes`. After `dtypes` property fix, this code started to work incorrectly.